### PR TITLE
[flant-integration] Distroless

### PR DESCRIPTION
**Pull request by vitaliy-sn**
## Description

- grafana-agent and madison-proxy images are now based on a distroless image
- flant-pricing are now based on ALT Linux image

## What is the expected result?

<details><summary>Details</summary>

```shell
root@dev-master-0:~# kubectl -n d8-flant-integration get po
NAME            READY   STATUS    RESTARTS   AGE
pricing-pdwh7   3/3     Running   0          12m
root@dev-master-0:~# kubectl -n d8-flant-integration logs pricing-pdwh7 pricing | head
time="2024-01-18T07:19:50Z" level=info msg="shell-operator v1.2.0-dev"
time="2024-01-18T07:19:50Z" level=info msg="Debug endpoint listen on /tmp/shell-operator-debug.socket"
time="2024-01-18T07:19:50Z" level=info msg="Listen on 0.0.0.0:9115"
time="2024-01-18T07:19:50Z" level=info msg="Create metric histogram shell_operator_kubernetes_client_request_latency_seconds" operator.component=metricsStorage
time="2024-01-18T07:19:50Z" level=info msg="Create metric counter shell_operator_kubernetes_client_request_result_total" operator.component=metricStorage
time="2024-01-18T07:19:50Z" level=info msg="Kubernetes client is configured successfully with 'out-of-cluster' config" operator.component=KubernetesAPIClient
time="2024-01-18T07:19:50Z" level=info msg="Kubernetes client is configured successfully with 'out-of-cluster' config" operator.component=KubernetesAPIClient
time="2024-01-18T07:19:50Z" level=info msg="Create metric counter shell_operator_live_ticks" operator.component=metricStorage
time="2024-01-18T07:19:50Z" level=info msg="Create metric histogram shell_operator_tasks_queue_action_duration_seconds" operator.component=metricsStorage
time="2024-01-18T07:19:50Z" level=info msg="Create metric gauge shell_operator_tasks_queue_length" operator.component=metricStorage
root@dev-master-0:~# kubectl -n d8-flant-integration logs pricing-pdwh7 grafana-agent | head
ts=2024-01-18T07:19:51.432537262Z level=info agent=prometheus component=cluster msg="applying config"
ts=2024-01-18T07:19:51Z level=info msg="Tempo Logger Initialized" component=tempo
ts=2024-01-18T07:19:51.528176386Z level=info agent=prometheus instance=f008ecadbddcb2dd42e5812f87557b46 msg="replaying WAL, this may take a while" dir=/data/agent/wal/f008ecadbddcb2dd42e5812f87557b46/wal
ts=2024-01-18T07:19:51.529591851Z level=info agent=prometheus instance=f008ecadbddcb2dd42e5812f87557b46 msg="WAL checkpoint loaded"
ts=2024-01-18T07:19:51.532313642Z level=info agent=prometheus component=cluster msg="not watching the KV, none set"
ts=2024-01-18T07:19:51.532637989Z level=info msg="server configuration changed, restarting server"
ts=2024-01-18T07:19:51.533046749Z level=info caller=server.go:245 http=[::]:8080 grpc=[::]:9095 msg="server listening on addresses"
ts=2024-01-18T07:19:51.536822119Z level=info agent=prometheus instance=f008ecadbddcb2dd42e5812f87557b46 msg="WAL segment loaded" segment=22 maxSegment=26
ts=2024-01-18T07:19:51.537269119Z level=info agent=prometheus instance=f008ecadbddcb2dd42e5812f87557b46 msg="WAL segment loaded" segment=23 maxSegment=26
ts=2024-01-18T07:19:51.538618078Z level=info agent=prometheus instance=f008ecadbddcb2dd42e5812f87557b46 msg="WAL segment loaded" segment=24 maxSegment=26
```

</details>

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: flant-integration
type: feature
summary: flant-pricing based on ALT Linux image, grafana-agent and madison-proxy images based on a distroless image.
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/static'...

File: ee/modules/600-flant-integration/images/flant-pricing/werf.inc.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  add: /shell-operator/shell_lib.sh
+  to: /shell_lib.sh
+  before: setup
+- artifact: {{ $.ModuleName }}/{{ $.ImageName }}-artifact
+  add: /shell-operator/frameworks
+  to: /frameworks
+  before: setup
+- artifact: {{ $.ModuleName }}/{{ $.ImageName }}-artifact
+  add: /jq/jq
+  to: /usr/local/bin/jq
```
Comment (RomanenkoDenys) [inline ee/modules/600-flant-integration/images/flant-pricing/werf.inc.yaml:21]: we have prebuilt common/shell-operator image

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
some nits

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/static'...

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.25 7672834068 7672834068-1-con-1-25 /sys/deckhouse-oss/install:pr6957 astra@192.168.199.125

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

File: ee/modules/600-flant-integration/images/flant-pricing/werf.inc.yaml (hunk 1 above)
Comment (vitaliy-sn) [reply ee/modules/600-flant-integration/images/flant-pricing/werf.inc.yaml]: done

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

**Comment by z9r5** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 7693041560 7693041560-1-con-1-25 /sys/deckhouse-oss/install:pr6957

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/static'...

**Comment by vitaliy-sn** [comment]
/e2e/abort static;Static;containerd;1.25 7737674025 7737674025-1-con-1-25 /sys/deckhouse-oss/install:pr6957 astra@192.168.199.202

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/static'...

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.25 7738648896 7738648896-1-con-1-25 /sys/deckhouse-oss/install:pr6957 astra@192.168.199.58

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/static'...

File: ee/be/modules/600-flant-integration/images/madison-proxy/werf.inc.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+---
+artifact: {{ .ModuleName }}/madison-proxy-artifact
+from: {{ .Images.BASE_GOLANG_21_ALPINE }}
+git:
+- add: /{{ $.ModulePath }}/modules/600-{{ $.ModuleName }}/images/{{ $.ImageName }}/src
+  to: /src
+  stageDependencies:
+    install:
+      - '**/*'
+shell:
```
Comment (RomanenkoDenys) [inline ee/be/modules/600-flant-integration/images/madison-proxy/werf.inc.yaml:10]:
```suggestion
mount:
- fromPath: ~/go-pkg-cache
  to: /go/pkg
shell:
```

File: ee/be/modules/600-flant-integration/images/grafana-agent/werf.inc.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+---
+artifact: {{ .ModuleName }}/agent-artifact
+from: {{ .Images.BASE_GOLANG_21_ALPINE }}
+shell:
```
Comment (RomanenkoDenys) [inline ee/be/modules/600-flant-integration/images/grafana-agent/werf.inc.yaml:4]:
```suggestion
mount:
- fromPath: ~/go-pkg-cache
  to: /go/pkg
shell:
```

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
some nits

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

**Review by RomanenkoDenys — APPROVED** [review]

**Pull request merged**